desktop proxies: clarify situation on Mac and Windows (#15045)
docker/docker.github.io: desktop/mac/index.md

#### Proxies

HTTP/HTTPS proxies can be used when

> - logging in to Docker

Reviewer: Nit pick: can we capitalize the first word of each bullet point to align with our style guide?

- pulling or pushing images
- fetching artifacts during image builds
- containers interact with the external network

> - scanning images.

Reviewer: ```suggestion
- scanning images
```

These are configured slightly differently.

> If the host uses a static HTTP/HTTPS proxy configuration then Docker Desktop reads this configuration

Reviewer: ```suggestion
If the host uses a static HTTP/HTTPS proxy configuration, Docker Desktop reads this configuration
```

and automatically uses these settings for logging into Docker and for pulling and pushing images.
If the host uses a more sophisticated HTTP/HTTPS configuration then enable "Manual proxy configuration"

> in the Settings / Resources / Proxies in the Docker Desktop UI and enter a single upstream proxy URL

Reviewer: ```suggestion
in the **Settings > Resources > Proxies** in Docker Dashboard and enter a single upstream proxy URL
```

of the form `http://username:password@proxy:port`.

HTTP/HTTPS traffic from image builds and running containers is forwarded transparently to the same
upstream proxy used for logging in and image pulls.

> If you wish to override this behaviour and use different HTTP/HTTPS proxies for image builds and

Reviewer: ```suggestion
If you want to override this behaviour and use different HTTP/HTTPS proxies for image builds and
```

> running containers then see [Configure the Docker client](/network/proxy#configure-the-docker-client).

Reviewer: ```suggestion
running containers, see [Configure the Docker client](/network/proxy#configure-the-docker-client).
```
